xds/internal/balancer/lrs/lrs.go

```diff
@@ -280,10 +282,12 @@ func (ls *lrsStore) buildStats(clusterName string) []*endpointpb.ClusterStats {
 	return ret
 }
 
+const nodeMetadataHostnameKey = "PROXYLESS_CLIENT_HOSTNAME"
+
 // ReportTo makes a streaming lrs call to cc and blocks.
 //
 // It retries the call (with backoff) until ctx is canceled.
-func (ls *lrsStore) ReportTo(ctx context.Context, cc *grpc.ClientConn, clusterName string, node *corepb.Node) {
+func (ls *lrsStore) ReportTo(ctx context.Context, cc *grpc.ClientConn, clusterName string, hostname string, node *corepb.Node) {
```

Comment:
Why does hostname refer to the user's dial target and not the hostname of the client. Sighhh.

Reply:
It's the parent ClientConn's hostname, not xds client's hostname.

I will need to do a big refactor of lrs, and move it to the xds client (to support reporting loads for multiple clusters).
The parent hostname will become a field of the xds client, so it's less weird. This also makes it impossible to reuse the same xds client across (parent) ClientConns.